petergreve/week2/Ass3/index.html

```diff
@@ -0,0 +1,233 @@
+					     					{message.text}
+					     					<button type="button" onClick={that.handleUnarchiveClick.bind(null, message)}>Unarchive</button>
+					     				</div>;
+					     	})}
+					    </ol>
+      				</div>
+		}
+
+	});
+
+	var ErrorMessage = React.createClass({
+		render: function () {
+
+			var style = {
+	   			color: 'red',
+    		};
+			return 	<div style={style}>
+						{this.props.errorMessage}
+					</div>
+		}
+
+	});
+
+	var App = React.createClass({
+		getInitialState: function() {
+		    return {
+		    		listCount: 2,
+		    		lists: [{listId: 0, listName: "first list"},{listId: 1, listName: "second list"}],
+		    		messageCount: 4,
+		    		errorMessage: '',
+		    		allMessages: [
+		    						{messageId: 0, listId: 0, text: "testmessage for list 0", isArchived: false},
+		    						{messageId: 3, listId: 0, text: "testmessage for list 0", isArchived: true},
+		    					 	{messageId: 1, listId: 1, text: "testmessage for list 1", isArchived: false},
+									{messageId: 2 ,listId: 2, text: "testmessage for list 2", isArchived: false}
+		    					 ]
+		    	};
+		},
+		render: function() {
+			var that = this;
+			return 	<div>
+		    			<input type="text" ref={(component) => this.newListName = component} />
+						<button type="button" onClick={this.createList}>New List</button>
+						<ol>
+					   		{this.state.lists.map(function(list, i) {
+					    		return <List
+					    					key={i}
+					    					handleInput={that.handleInput}
+					    					archiveMessage={that.archiveMessage}
+					    					createMessage={that.createMessage}
+					    					deleteMessage={that.deleteMessage}
+					    					messages={this.filterMessages(list.listId)}
+					    					archived={this.filterArchived(list.listId)}
+					    					moveMessage={that.moveMessage}
+					    					list={list} />;
+					   		},this)}
+					    </ol>			
+					</div>
+		},
+		handleInput: function(message) {
+			if (message.length<200) {
+				this.setState({
+		          errorMessage: '',
+		          allMessages: this.state.allMessages.concat([message]),
+		      	});
+			} else {
+				this.setState({
+		          errorMessage: message,
+				});
+			}
+		},
+		createList: function() {
+			var newList = [{
+				listId: this.state.listCount,
+				listName: this.newListName.value
+			}];
+
+			this.setState({
+		          lists: this.state.lists.concat(newList),
+		          listCount: this.state.listCount + 1
+		      	});
+		},
+		createMessage: function(text, listId) {
+			if (text.length<200) {
+				var newMessage = [{
+					messageId: this.state.messageCount,
+					text: text,
+					isArchived: false,
+					listId: listId
+				}];
+
+				this.setState({
+			          allMessages: this.state.allMessages.concat(newMessage),
+			          messageCount: this.state.messageCount + 1
+			      	});
+			} 
+		},
+		filterMessages: function(listId) {
+			return this.state.allMessages.filter((m) => {
+				return m.isArchived == false && m.listId == listId;
```

Comment:
it depends on religion, but you could write `!m.isArchived`, but it do seem more readable this way